### PR TITLE
Update Azure Identity docs for new azd credential

### PR DESCRIPTION
**Pull request by scottaddie**
Addresses https://github.com/Azure/azure-sdk-for-net/pull/33428#issuecomment-1403199778

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

**Comment by vhvb1989** [comment]
Do we have a similar PR for JS, JAVA and Py?

File: sdk/identity/Azure.Identity/README.md [hunk 1]
```diff
@@ -211,6 +217,7 @@ Not all credentials require this configuration. Credentials which authenticate t
 |Credential | Usage | Reference
 |-|-|-
 |[`AzureCliCredential`][ref_AzureCliCredential]|Authenticates in a development environment with the Azure CLI. | [Azure CLI authentication](https://learn.microsoft.com/cli/azure/authenticate-azure-cli)
+|`AzureDeveloperCliCredential`|Authenticates in a development environment with the Azure Developer CLI. | N/A
```
Comment (schaabs) [inline sdk/identity/Azure.Identity/README.md:220]:
```suggestion
|`AzureDeveloperCliCredential`|Authenticates in a development environment with the Azure Developer CLI. | [Azure Developer CLI Reference](https://learn.microsoft.com/azure/developer/azure-developer-cli/reference?source=recommendations)
```

**Review by schaabs — APPROVED** [review]

**Comment by scottaddie** [comment]
> Do we have a similar PR for JS, JAVA and Py?
Not yet. I've asked @KarishmaGhiya, @g2vinay, and @xiangyan99 to take care of that.

File: sdk/identity/Azure.Identity/README.md [hunk 2]
```diff
@@ -83,12 +86,13 @@ The `DefaultAzureCredential` attempts to authenticate via the following mechanis
...
+1. **Azure Developer CLI** - If the developer has authenticated via the Azure Developer CLI `azd login` command, the `DefaultAzureCredential` will authenticate with that account.
+1. **Visual Studio** - If the developer has authenticated via Visual Studio, the `DefaultAzureCredential` will authenticate with that account.
+1. **Visual Studio Code** - Currently excluded by default as SDK authentication via Visual Studio Code is broken due to issue [#27263](https://github.com/Azure/azure-sdk-for-net/issues/27263). The `VisualStudioCodeCredential` will be re-enabled in the `DefaultAzureCredential` flow once a fix is in place. Issue [#30525](https://github.com/Azure/azure-sdk-for-net/issues/30525) tracks this. In the meantime Visual Studio Code users can authenticate their development environment using the [Azure CLI](https://learn.microsoft.com/cli/azure/).
+1. **Azure CLI** - If the developer has authenticated an account via the Azure CLI `az login` command, the `DefaultAzureCredential` will authenticate with that account.
```
Comment (billwert) [inline sdk/identity/Azure.Identity/README.md:93]: should azd have gone here?

Comment (scottaddie) [reply sdk/identity/Azure.Identity/README.md:93]: It's on line 90

**Pull request merged**